bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs

```diff
@@ -23,14 +24,47 @@ namespace MongoDB.Libmongocrypt
     /// </summary>
     public class CryptClientFactory
     {
+        // MUST be static fields since otherwise these callbacks can be collected via the garbage collector
+        // regardless they're used by mongocrypt level or no
+        private static Library.Delegates.HashCallback __cryptoHashCallback = new Library.Delegates.HashCallback(HashCallback.Hash);
+        private static Library.Delegates.CryptoCallback __crypto256EncryptCallback = new Library.Delegates.CryptoCallback(CipherCallbacks.Encrypt);
+        private static Library.Delegates.CryptoCallback __crypto256DecryptCallback = new Library.Delegates.CryptoCallback(CipherCallbacks.Decrypt);
+        private static Library.Delegates.RandomCallback __randomCallback = new Library.Delegates.RandomCallback(SecureRandomCallback.GenerateRandom);
+        private static Library.Delegates.CryptoHmacCallback __cryptoHmacSha512Callback = new Library.Delegates.CryptoHmacCallback(HmacShaCallbacks.HmacSha512);
+        private static Library.Delegates.CryptoHmacCallback __cryptoHmacSha256Callback = new Library.Delegates.CryptoHmacCallback(HmacShaCallbacks.HmacSha256);
+
         /// <summary>Creates a CryptClient with the specified options.</summary>
         /// <param name="options">The options.</param>
         /// <returns>A CryptClient</returns>
         public static CryptClient Create(CryptOptions options)
         {
-            MongoCryptSafeHandle handle = Library.mongocrypt_new();
+            var handle = Library.mongocrypt_new();
+
+            var status = new Status();
+
+            // The below code can be avoided on Windows. So, we don't call it on this system 
+            // to avoid restrictions on target frameworks that present in some of below
+            //if (OperatingSystemHelper.CurrentOperatingSystem != OperatingSystemPlatform.Windows)
```

Comment:
this will be restored. Need to be able making testing on windows as well

Reply:
Can you rephrase the second sentence of your comment?  I'm not able to parse it.

Reply:
Sorry, I meant the second sentence of the PR comment!  I don't know what "Need to be able making testing on windows as well" means.  What do you have to do before uncommenting the line of code?

Reply:
> What do you have to do before uncommenting the line of code?

nothing, but if we uncomment this line, the callbacks won't be called (since I test only on windows for now) at all. So I made this temporal change just to show how it can be checked on windows if someone wants to try it